HADOOP-19105. s3A: Improve resilience in vector reads.

steveloughran opened the pull request:
Add a new releaser method, which then is invoked to release buffers on failure.

It is a bit contrived how we try not break external implementations when adding a new default implementation to PositionedReadable: the releaser will be lost unless they do so.





### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

steveloughran committed:
HADOOP-19105. Improve resilience in vector reads.

Add a new releaser method, which then is invoked to release
buffers on failure.

It is a bit contrived how we try not break external
implementations when adding a new default implementation
to PositionedReadable: the releaser will be lost unless
they do so.

Change-Id: Ie2f95a89cfa79d3fdf33c1cc58f63e4ca00c6207

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  11m 55s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 57s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  31m 41s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  17m 46s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  compile  |  16m 26s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  checkstyle  |   4m 21s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 42s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m 14s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javadoc  |   1m 42s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   3m 56s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  35m 47s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 30s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m  4s |  |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javac  |  17m  4s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 12s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  javac  |  16m 12s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   4m 13s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/1/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 2 new + 57 unchanged - 0 fixed = 59 total (was 57)  |
| +1 :green_heart: |  mvnsite  |   2m 41s |  |  the patch passed  |
| -1 :x: |  javadoc  |   1m 15s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/1/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04.txt) |  hadoop-common-project_hadoop-common-jdkUbuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  javadoc  |   1m 46s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   4m 18s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  36m  2s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  19m 48s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 56s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  5s |  |  The patch does not generate ASF License warnings.  |
|  |   | 258m  1s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7105 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux c313793e4f5e 5.15.0-119-generic #129-Ubuntu SMP Fri Aug 2 19:25:20 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 311c2e09995c8a82440024d082d87719333e3171 |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/1/testReport/ |
| Max. process+thread count | 3152 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran committed:
HADOOP-19105. Improve resilience in vector reads.

Trying to implement resilience in the s3a input stream;
buffers of incomplete reads are now released on failures.

Change-Id: I4b23c6a2ff417687fd19c1e8e9feef6cf733c031

steveloughran committed:
HADOOP-19105. Improve resilience in vector reads.

Trying to implement resilience in the s3a input stream;
buffers of incomplete reads are now released on failures.

Change-Id: I851779383f8d120fe42bc97c5e6ae533c5fe2e23

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 48s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  32m 41s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  18m  1s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  compile  |  16m 55s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  checkstyle  |   4m 24s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 45s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m 10s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javadoc  |   1m 47s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   4m  0s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  35m 55s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 30s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m  4s |  |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javac  |  17m  4s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 23s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  javac  |  16m 23s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   4m 22s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/2/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 2 new + 57 unchanged - 0 fixed = 59 total (was 57)  |
| +1 :green_heart: |  mvnsite  |   2m 39s |  |  the patch passed  |
| -1 :x: |  javadoc  |   1m 14s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/2/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04.txt) |  hadoop-common-project_hadoop-common-jdkUbuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  javadoc  |   1m 48s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   4m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  36m 28s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  20m  2s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  5s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  7s |  |  The patch does not generate ASF License warnings.  |
|  |   | 249m 49s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7105 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux bd03228d87db 5.15.0-117-generic #127-Ubuntu SMP Fri Jul 5 20:13:28 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 29e20495659a61ab84db2aed1d0553151818f87a |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/2/testReport/ |
| Max. process+thread count | 1251 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran committed:
HADOOP-19105. s3a input stream codepath looks good.

Now: how do I pull this out so it can be used with the prefetch
stream?

Change-Id: I98e350ab3b28d9adc7b458dee4ceee56a0562f17

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 49s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  31m 36s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  17m 45s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  compile  |  16m 42s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  checkstyle  |   4m 23s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 41s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m 12s |  |  trunk passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javadoc  |   1m 47s |  |  trunk passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   3m 57s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  35m 40s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  36m  6s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 27s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 28s |  |  the patch passed with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04  |
| +1 :green_heart: |  javac  |  17m 28s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 44s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  javac  |  16m 44s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   4m 12s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/3/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 2 new + 57 unchanged - 0 fixed = 59 total (was 57)  |
| +1 :green_heart: |  mvnsite  |   2m 38s |  |  the patch passed  |
| -1 :x: |  javadoc  |   1m 12s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/3/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04.txt) |  hadoop-common-project_hadoop-common-jdkUbuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 with JDK Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  javadoc  |   1m 46s |  |  the patch passed with JDK Private Build-1.8.0_422-8u422-b05-1~20.04-b05  |
| +1 :green_heart: |  spotbugs  |   4m 20s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  35m 39s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  19m 34s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  0s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  6s |  |  The patch does not generate ASF License warnings.  |
|  |   | 247m 32s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7105 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 1c492766ce40 5.15.0-119-generic #129-Ubuntu SMP Fri Aug 2 19:25:20 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / fef176208ad4dbe6153a1944ddb545204c7a6537 |
| Default Java | Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.24+8-post-Ubuntu-1ubuntu320.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_422-8u422-b05-1~20.04-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/3/testReport/ |
| Max. process+thread count | 3152 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran committed:
Merge branch 'trunk' into s3/HADOOP-19105-vectorio-recovery

Conflicts:
  hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AInputStream.java

Change-Id: I46473271c9fdc9c5b3bbc561ecdc0a4ea36988ce

steveloughran committed:
Merge branch 'trunk' into s3/HADOOP-19105-vectorio-recovery

Change-Id: I669d2bf55795a091c0d2a3b54a11cd85dc2cee1c

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   7m 23s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  30m 22s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  16m 26s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  15m 20s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  checkstyle  |   4m 13s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 34s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m  9s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 44s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  spotbugs  |   3m 49s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  33m 42s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  34m  8s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 24s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 43s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  15m 43s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m  4s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  javac  |  15m  4s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   4m 13s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/4/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 2 new + 57 unchanged - 0 fixed = 59 total (was 57)  |
| +1 :green_heart: |  mvnsite  |   2m 34s |  |  the patch passed  |
| -1 :x: |  javadoc  |   1m 12s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/4/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04.txt) |  hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  javadoc  |   1m 45s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  spotbugs  |   4m  5s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  33m 41s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  19m 57s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  7s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  4s |  |  The patch does not generate ASF License warnings.  |
|  |   | 227m 40s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7105 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 613e1b441543 5.15.0-124-generic #134-Ubuntu SMP Fri Sep 27 20:20:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 2e8b7dcedf1f9a9dc576911841c1bd8893529736 |
| Default Java | Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/4/testReport/ |
| Max. process+thread count | 2353 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/4/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran committed:
HADOOP-19105. Improve resilience in vector reads.

Rigorous aborting of streams where appropriate, and
if one range read fails in the fallback sequential read,
so do all the others.

factor out start/stop.
This  is preparing to pull this into a superclass so that all s3a
streams get it.

Change-Id: I8961d6737764bb696aa45338bb0300b2f6efeec6

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 33s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 49s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  30m 12s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  16m 21s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  15m  0s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  checkstyle  |   4m 10s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   3m 28s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   3m  0s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   2m 32s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  spotbugs  |   5m  4s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  33m 26s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  33m 52s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 56s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 44s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  15m 44s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 21s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  javac  |  15m 21s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   4m  8s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/5/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 3 new + 58 unchanged - 0 fixed = 61 total (was 58)  |
| +1 :green_heart: |  mvnsite  |   3m 26s |  |  the patch passed  |
| -1 :x: |  javadoc  |   1m 10s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/5/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04.txt) |  hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  javadoc  |   2m 34s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| -1 :x: |  spotbugs  |   1m 34s | [/new-spotbugs-hadoop-tools_hadoop-aws.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/5/artifact/out/new-spotbugs-hadoop-tools_hadoop-aws.html) |  hadoop-tools/hadoop-aws generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  shadedclient  |  33m 59s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  19m 50s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  8s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 40s |  |  hadoop-azure in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  3s |  |  The patch does not generate ASF License warnings.  |
|  |   | 237m 47s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-tools/hadoop-aws |
|  |  Exceptional return value of java.util.concurrent.ExecutorService.submit(Callable) ignored in org.apache.hadoop.fs.s3a.S3AInputStream.readVectored(List, IntFunction, Consumer)  At S3AInputStream.java:ignored in org.apache.hadoop.fs.s3a.S3AInputStream.readVectored(List, IntFunction, Consumer)  At S3AInputStream.java:[line 1023] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7105 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets markdownlint |
| uname | Linux 196697e78f52 5.15.0-124-generic #134-Ubuntu SMP Fri Sep 27 20:20:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 13e168e98174d8cd9f6d4fdbe31897d48631199a |
| Default Java | Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/5/testReport/ |
| Max. process+thread count | 2155 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws hadoop-tools/hadoop-azure U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/5/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
this is really complicated

* going to pull out the vector minimum ranges commit to its own
* maybe move the api change in first
* isolate all s3a vector io support into its own class, to work with any object input streams, so adds it to analytics stream too

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 31s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   7m 34s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  29m 51s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  16m 42s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  14m 58s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  checkstyle  |   4m 13s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   3m 32s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m 59s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   2m 33s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  spotbugs  |   5m  7s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  34m 10s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  34m 36s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 54s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 52s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  15m 52s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 12s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  javac  |  15m 12s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   4m 14s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/6/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 3 new + 58 unchanged - 0 fixed = 61 total (was 58)  |
| +1 :green_heart: |  mvnsite  |   3m 28s |  |  the patch passed  |
| -1 :x: |  javadoc  |   1m 11s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/6/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04.txt) |  hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  javadoc  |   2m 33s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| -1 :x: |  spotbugs  |   1m 32s | [/new-spotbugs-hadoop-tools_hadoop-aws.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/6/artifact/out/new-spotbugs-hadoop-tools_hadoop-aws.html) |  hadoop-tools/hadoop-aws generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  shadedclient  |  34m  1s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  19m 55s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  7s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 41s |  |  hadoop-azure in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  5s |  |  The patch does not generate ASF License warnings.  |
|  |   | 239m 44s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-tools/hadoop-aws |
|  |  Exceptional return value of java.util.concurrent.ExecutorService.submit(Callable) ignored in org.apache.hadoop.fs.s3a.S3AInputStream.readVectored(List, IntFunction, Consumer)  At S3AInputStream.java:ignored in org.apache.hadoop.fs.s3a.S3AInputStream.readVectored(List, IntFunction, Consumer)  At S3AInputStream.java:[line 1021] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7105 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets markdownlint |
| uname | Linux cf631fdf5960 5.15.0-124-generic #134-Ubuntu SMP Fri Sep 27 20:20:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 01b421955776b48e24425aef72b16fcce982dbe7 |
| Default Java | Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/6/testReport/ |
| Max. process+thread count | 1289 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws hadoop-tools/hadoop-azure U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/6/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 26s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  29m 57s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  16m 29s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  15m  3s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  checkstyle  |   4m 19s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 34s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m 10s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 44s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  spotbugs  |   3m 51s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  33m 29s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  33m 54s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 25s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 52s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  15m 52s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  14m 59s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  javac  |  14m 59s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   4m  8s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/7/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 2 new + 57 unchanged - 0 fixed = 59 total (was 57)  |
| +1 :green_heart: |  mvnsite  |   2m 29s |  |  the patch passed  |
| -1 :x: |  javadoc  |   1m 11s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/7/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04.txt) |  hadoop-common-project_hadoop-common-jdkUbuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  javadoc  |   1m 43s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| -1 :x: |  spotbugs  |   1m 33s | [/new-spotbugs-hadoop-tools_hadoop-aws.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/7/artifact/out/new-spotbugs-hadoop-tools_hadoop-aws.html) |  hadoop-tools/hadoop-aws generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  shadedclient  |  35m  4s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  19m 54s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  6s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  3s |  |  The patch does not generate ASF License warnings.  |
|  |   | 227m  8s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-tools/hadoop-aws |
|  |  Exceptional return value of java.util.concurrent.ExecutorService.submit(Callable) ignored in org.apache.hadoop.fs.s3a.S3AInputStream.readVectored(List, IntFunction, Consumer)  At S3AInputStream.java:ignored in org.apache.hadoop.fs.s3a.S3AInputStream.readVectored(List, IntFunction, Consumer)  At S3AInputStream.java:[line 1021] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7105 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux db64612c0853 5.15.0-124-generic #134-Ubuntu SMP Fri Sep 27 20:20:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / e11b0a9dafdbeb823d5d7f2a5bdd699dd1aadb2f |
| Default Java | Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/7/testReport/ |
| Max. process+thread count | 3154 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7105/7/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.
